internal/pkg/node/node.go

```diff
@@ -436,34 +470,43 @@ func (node *Node) checkPredecessor() {
 
 	if err != nil {
 		node.predecessor = nil
+		predecessor.CloseConnection()
+		delete(remotes.data, predecessor.GetAddr())
 	}
 }
 
 // Notify is called when another node thinks it is our predecessor
 func (node *Node) Notify(id scale.Key, addr string) error {
-	node.mutex.Lock()
-	defer node.mutex.Unlock()
 
```

Comment:
removed the Lock() defer Unlock() because when `node.bootstrap()` was called it caused a deadlock.

Reply:
ayyyy fun times